File: 128-Longest-Consecutive-Sequence/solution.java
public class Solution {
    public int longestConsecutive(int[] nums) {
        HashSet<Integer> set = new HashSet<>();
        for(Integer num: nums){
            set.add(num);
        }
        int longest = 0;
        for(int i =0; i < nums.lenght; i++){
            int down = nums[i] -1;
            while(set.contains(down)){
                set.remove(down);
                down--;
            }
            
            int up = nums[i] + 1;
            while(set.contains(up)){
                set.remove(up);
                up++;
            }
            longest = Math.max(longest, up-down-1);
        }
        return longest;
    }
}

File: 128-Longest-Consecutive-Sequence/solution.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestConsecutive = function(nums) {
    var map = {};
    nums.forEach(function(num){
        map[num] = 1;
    });
    var result = 0;
    nums.forEach(function(num){
        var up = num + 1;
        while(map[up]){
            delete map[up];
            up++;
        }
        
        var down = num -1;
        while(down){
             delete map[down];
             down--;
        }
        console.log('up', up);
        console.log('down', down);
        result = Math.max(result, up-down-1);
    });
    return result;
    
};